solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -865,6 +871,76 @@ public void removeLiveNodesListener(LiveNodesListener listener) {
     liveNodesListeners.remove(listener);
   }
 
+  /**
+   * Returns the lowest Solr version among all live nodes in the cluster.
+   *
+   * @return the lowest Solr version as a String, or null if no version is found
+   * @throws KeeperException if a ZooKeeper error occurs
+   * @throws InterruptedException if the thread is interrupted
+   */
+  public String getLowestSolrVersion() throws KeeperException, InterruptedException {
```

Comment:
I personally hate it when a "get" method does network access; it's so subtle -- the caller would have no clue except maybe the exceptions that can be thrown as seen here.  Using a different verb like "fetch" would be suitable.  But actually what I expected/hoped was that we'd cache the live node metadata in ZkStateReader from live node watching.  I could imagine this might be more scope than you had in mind; maybe this here is fine for now.